src/tools/perf/lib/uct_tests.cc

```diff
@@ -188,7 +189,11 @@ class uct_perf_test_runner {
         case UCX_PERF_CMD_PUT:
             if (TYPE == UCX_PERF_TEST_TYPE_PINGPONG) {
                 /* Put the control word at the latest byte of the IOV message */
-                *((psn_t*)buffer + uct_perf_get_buffer_extent(&m_perf.params) - 1) = sn;
+                m_perf.allocator->memcpy(((psn_t*)buffer +
+                                          uct_perf_get_buffer_extent(&m_perf.params) - 1),
```

Comment:
i think we better use UCS_PTR_BYTE_OFFSET for uct_perf_get_buffer_extent:
`(psn_t*)UCS_PTR_BYTE_OFFSET(buffer, uct_perf_get_buffer_extent(&m_perf.params) - ``
and to make it readable, split to a helper variable "psn_t *psn_ptr" or similar